src/query/execute_schema.c

```diff
@@ -10016,6 +10020,366 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+			      alter->info.alter.constraint_list, &attr_chg_prop, &change_mode);
+
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+
+  /* save class MOP */
+  class_mop = ctemplate->op;
+
+  /* check foreign key constraints */
+  error = do_check_fk_constraints (ctemplate, alter->info.alter.constraint_list);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+
+  is_srv_update_needed = ((change_mode == SM_ATTR_CHG_WITH_ROW_UPDATE || change_mode == SM_ATTR_CHG_BEST_EFFORT)
+			 && attr_chg_prop.name_space == ID_ATTRIBUTE) ? true : false;
+  if (is_srv_update_needed)
+    {
+      COPY_OID (&class_oid, &(ctemplate->op->oid_info.oid));
+      att_id = attr_chg_prop.att_id;
+    }
+
+  /* force schema update to server */
+  class_obj = dbt_finish_class (ctemplate);
+  if (class_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto exit;
+    }
+  /* set NULL, avoid 'abort_class' in case of error */
+  ctemplate = NULL;
+
+  if (attr_chg_prop.constr_info != NULL)
+    {
+      SM_CONSTRAINT_INFO *saved_constr = NULL;
+
+      for (saved_constr = attr_chg_prop.constr_info; saved_constr != NULL; saved_constr = saved_constr->next)
+       {
+	 if (saved_constr->func_index_info || saved_constr->filter_predicate)
+	   {
+	     if (saved_constr->func_index_info)
+	       {
+		 error = do_recreate_func_index_constr (parser, saved_constr, NULL, alter, NULL, NULL);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+	       }
+	     if (saved_constr->filter_predicate)
+	       {
+		 error = do_recreate_filter_index_constr (parser, saved_constr->filter_predicate, alter, NULL, NULL);
+		 if (error != NO_ERROR)
+		   {
+		     goto exit;
+		   }
+	       }
+
+	     if (!is_srv_update_needed)
+	       {
+		 const char *att_names[2];
+		 PT_NODE *att_old_name = alter->info.alter.alter_clause.attr_mthd.attr_old_name;
+
+		 if (att_old_name != NULL)
+		   {
+		     assert (att_old_name->node_type == PT_NAME);
+		     att_names[0] = att_old_name->info.name.original;
+		     att_names[1] = NULL;
+
+		     assert (alter->info.alter.alter_clause.attr_mthd.attr_old_name->node_type == PT_NAME);
+		     error =
+		       sm_drop_constraint (class_mop, saved_constr->constraint_type, saved_constr->name, att_names,
+					   false, false);
+
+		     if (error != NO_ERROR)
+		       {
+			 goto exit;
+		       }
+
+		     error = sm_add_constraint (class_mop, saved_constr->constraint_type, saved_constr->name,
+						(const char **) saved_constr->att_names, saved_constr->asc_desc,
+						saved_constr->prefix_length, false, saved_constr->filter_predicate,
+						saved_constr->func_index_info, saved_constr->comment,
+						saved_constr->index_status);
+		     if (error != NO_ERROR)
+		       {
+			 goto exit;
+		       }
+		   }
+	       }
+	   }
+       }
+    }
+
+  if (is_srv_update_needed || is_att_prop_set (attr_chg_prop.p[P_TYPE], ATT_CHG_TYPE_PREC_INCR))
+    {
```

Comment:
**ATT_CHG_TYPE_PREC_INCR** is set at altering precision of type, so I think that this conditional codes are not necessesary.

Reply:
Thank you. The code has been removed at the final commit.